.github/workflows/clang-tidy.yml

```diff
@@ -32,18 +43,30 @@ jobs:
       CXX: clang++-14
       CLANG_TIDY: clang-tidy-14
       PRESET: ci-clang-tidy-${{ matrix.build-option }}
+      TARGET: power_grid_model_c
 
     steps:
       - uses: actions/checkout@v3
+
       - name: Install packages
         run: |
           sudo apt-get update
           sudo apt-get install -y ninja-build
+
       - name: Enable brew
         run: |
           echo "/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin" >> $GITHUB_PATH
+
       - name: Install C++ dependencies
         run: |
           brew install boost eigen nlohmann-json doctest
-      - name: Build and test
-        run: ./build.sh -p ${{ env.PRESET }}
+
+      - name: Override build target in case of workflow dispatch
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          echo "TARGET=${{ github.event.inputs.target }}"
+
+      - name: Configure
+        run: |
+          cmake --preset ${{ env.PRESET }}
+          cmake --build --preset ${{ env.PRESET }} --target ${{ env.TARGET }}
```

Comment:
I do not understand how you override the default target. In your current way, `${{ env.TARGET }}` is always `power_grid_model_c`

Reply:
unfortunately, I can only properly test that this works correctly after this is merged to main because github actions only uses the default branch to check for workflow dispatch triggers

Reply:
I doubt this will work as intended. Using the syntax you have.

You can test it in another way, Just do echo in one step and print the result in another step.

Reply:
ohhh i see i forgot the `>> $GITHUB_ENV`.

i will do a couple more tests

Reply:
ok it definitely works now. thanks